internal/controller/plugin/plugin_controller_flux.go

```diff
@@ -119,6 +119,7 @@ func (r *PluginReconciler) EnsureFluxCreated(ctx context.Context, plugin *greenh
 				Name: plugin.Spec.ClusterName,
 				Key:  greenhouseapis.GreenHouseKubeConfigKey,
 			}).
+			WithDependsOn(convertWaitForToDependsOn(plugin.Spec.WaitFor)).
```

Comment:
The WaitFor could still have both a Name and a PluginRef at this point (iff set on a Plugin directly) That means the conversion would silently drop the PluginPreset reference at this point.
At the top of my head I cannot come up with any dependencies where the plugin would have a PluginPreset specced in the waitFor.
Let's quickly discuss in slack but I think we could use different structs for PluginPreset and Plugin to make validation simpler and ensure that the `n:n` references are only speccable on the Preset and that the Plugin only has `1:1` via the name field.